packages/devextreme/js/__internal/core/state_manager/reactive_primitives/types.ts

```diff
@@ -0,0 +1,16 @@
+import type * as SignalsCore from '@preact/signals-core';
+
+export type Signal<T> = SignalsCore.Signal<T>;
+export type ReadonlySignal<T> = SignalsCore.ReadonlySignal<T>;
+
+export type EffectCleanup = () => void;
+
+// eslint-disable-next-line @typescript-eslint/no-invalid-void-type
+export type EffectFn = () => void | EffectCleanup;
```

Comment:
Let's fix linter error instead of disabling it

```suggestion
export type EffectFn = (() => void) | EffectCleanup;
```

Reply:
The eslint rule is not suitable in this context. 
The type should describe returning void or a function returning void.
https://www.npmjs.com/package/@preact/signals-core/v/1.8.0?activeTab=code
Same as here `/@preact/signals-core/dist/signals-core.d.ts`

Reply:
Ok, let's leave as is